### ISSUE TITLE
Making DataTable responsive when the parent width is smaller than the DataTable itself

### ISSUE THREAD

**Issue by khalyomede** [material-components/material-components-web#5000]
## Feature Request
Making DataTable to behave responsively when the parent element is smaller than the DataTable itself.

### Proposed solution
The solution I propose can be viewed in this JSFiddle: [https://jsfiddle.net/97ezgasp/3/](https://jsfiddle.net/97ezgasp/3/). If you do not have access to it, here is a preview of the rendered solution:

![issue-data-table-not-responsive](https://user-images.githubusercontent.com/15908747/63086427-2032cd80-bf50-11e9-87fd-3a0a7de5e64f.png)

It only adds additionnal styles to make the table stretch to a maximum of 100% of its parent width. As the content inside of the DataTable might still overflows its container (the DataTable itself), the style also use `overflow-y: auto` to make the DataTable scrollable if the content happens to overflows its DataTable. Here is an excerpt of the style if you cannot view the JSFiddle:

```css
.mdc-data-table-responsive {
  max-width: 100%;
  overflow-y: auto;
}
```

### Alternatives considered
I also considered using an `overflow-y: auto` on the parent of the DataTable (for example, the body of a Card), but as the issue came back, I prefered to propose a solution where we could eventually opt-in this behavior (like a configuration in the Javascript? or an additionnal class like I proposed).

### Additional context
As a matter of fact, I did not see any consideration on the responsiveness of the DataTable in the [core Material Design DataTable](https://material.io/design/components/data-tables.html#implementation) documentation. 

Should I start with issuing the core documentation? Or is this an issue related to the userland of  the web realm?

**Comment by abhiomkar** [comment]
Thanks for the suggestion.

Can you also mention MDC Web version that you've tried. There are some recent changes to Data Table might fix this.

**Comment by abhiomkar** [comment]
Fixed in #5037
